core/src/main/java/com/scalar/db/transaction/consensuscommit/MutationsGrouper.java

```diff
@@ -50,6 +51,41 @@ public List<List<Mutation>> groupMutations(Collection<Mutation> mutations)
     return groupToBatches.values().stream().flatMap(List::stream).collect(Collectors.toList());
   }
 
+  public boolean canBeGroupedTogether(Collection<Mutation> mutations) throws ExecutionException {
```

Comment:
This method determines whether all the mutations can be grouped into a single group.